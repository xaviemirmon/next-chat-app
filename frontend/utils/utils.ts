export const isOnlyEmoji = (message: string) => {
  const trimmedMessage = message.trim();
  return trimmedMessage && /^[\p{Emoji}]+$/u.test(trimmedMessage);
};
