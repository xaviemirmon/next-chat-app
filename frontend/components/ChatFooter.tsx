import { MessageType, UserType } from "@/types/types";
import { handleKeyDown } from "@/utils/utils";

import styles from "./ChatFooter.module.css";
import { Dispatch, MutableRefObject, SetStateAction } from "react";

export const ChatFooter = ({
  input,
  setInput,
  targetUserData,
  setMessages,
  ws,
  connectionId,
  user,
  target,
}: {
  input: string;
  setInput: Dispatch<SetStateAction<string>>;
  targetUserData: UserType | undefined;
  setMessages: Dispatch<SetStateAction<MessageType[]>>;
  ws: MutableRefObject<WebSocket | null>;
  connectionId: MutableRefObject<number | null>;
  user: number | null;
  target: number;
}) => {
  const sendMessage = () => {
    if (ws.current && input.trim() && user && connectionId.current) {
      const message: MessageType = {
        type: "message",
        senderId: user,
        targetId: target,
        content: input,
        connectionId: connectionId.current,
        createdAt: new Date().toISOString(),
      };

      setMessages((prevMessages: MessageType[]) => [
        ...prevMessages,
        { createdAt: new Date().toISOString(), ...message },
      ]);
      ws.current.send(JSON.stringify(message));
      setInput("");
    }
  };
  return (
    <div className={styles.footer}>
      <input
        type="text"
        value={input}
        className={styles.input}
        onChange={(e) => setInput(e.target.value)}
        placeholder={`Message ${targetUserData?.name}`}
        onKeyDown={(e) => handleKeyDown(e, sendMessage)}
      />
    </div>
  );
};
