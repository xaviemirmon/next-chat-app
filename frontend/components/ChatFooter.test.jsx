import { render } from "@testing-library/react";
import { ChatFooter } from "./ChatFooter";
import { handleKeyDown } from "@/utils/utils";

// Mocking the utility function
jest.mock("@/utils/utils", () => ({
  handleKeyDown: jest.fn(),
}));

describe("ChatFooter component", () => {
  const mockSetInput = jest.fn();
  const mockSetMessages = jest.fn();
  const mockWs = { current: { send: jest.fn() } };
  const mockConnectionId = { current: 123 };
  const userId = 1;
  const targetId = 2;

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("Matches snapshot when input is empty", () => {
    const { asFragment } = render(
      <ChatFooter
        input=""
        setInput={mockSetInput}
        targetUserData={{ name: "Target User", userId: targetId }}
        setMessages={mockSetMessages}
        ws={mockWs}
        connectionId={mockConnectionId}
        user={userId}
        target={targetId}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Matches snapshot when input has text", () => {
    const { asFragment } = render(
      <ChatFooter
        input="Hello"
        setInput={mockSetInput}
        targetUserData={{ name: "Target User", userId: targetId }}
        setMessages={mockSetMessages}
        ws={mockWs}
        connectionId={mockConnectionId}
        user={userId}
        target={targetId}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Matches snapshot when targetUserData is undefined", () => {
    const { asFragment } = render(
      <ChatFooter
        input="Hello"
        setInput={mockSetInput}
        targetUserData={undefined}
        setMessages={mockSetMessages}
        ws={mockWs}
        connectionId={mockConnectionId}
        user={userId}
        target={targetId}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
