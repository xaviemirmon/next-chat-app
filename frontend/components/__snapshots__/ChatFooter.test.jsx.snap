// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatFooter component Matches snapshot when input has text 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <input
      class="input"
      placeholder="Message Target User"
      type="text"
      value="Hello"
    />
  </div>
</DocumentFragment>
`;

exports[`ChatFooter component Matches snapshot when input is empty 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <input
      class="input"
      placeholder="Message Target User"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`ChatFooter component Matches snapshot when targetUserData is undefined 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <input
      class="input"
      placeholder="Message undefined"
      type="text"
      value="Hello"
    />
  </div>
</DocumentFragment>
`;
